### PR TITLE
dreddy-aws/resource-return-values

### PR DESCRIPTION
**Pull request by dreddy-aws**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: workshop/content/basics/templates/finding-output-values/_index.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+      ImageId: !Ref LatestAmiId
+      InstanceType: t2.micro
+      SecurityGroups:
+        - !Ref InstanceSecurityGroup
+      Tags:
+        - Key: Name
+          Value: Resource-dependencies-workshop
+```
+
+* Use `Ref` intrinsic function, and pass the logical ID of the Amazon EC2 instance resource to retrieve the instance ID; similarly, pass the logical ID of the Amazon EC2 instance resource along with the `PublicIp` attribute to `Fn::GetAtt` function, to retrieve the public IP of the instance.
```
Comment (mrinaudo-aws) [inline workshop/content/basics/templates/finding-output-values/_index.md:142]:
```suggestion
* Use `Ref` intrinsic function, and pass the logical ID of the Amazon EC2 instance resource to retrieve the instance ID; similarly, pass the logical ID of the Amazon EC2 instance resource, along with the `PublicIp` attribute to the `Fn::GetAtt` function, to retrieve the public IP of the instance.
```

File: workshop/content/basics/templates/finding-output-values/_index.md [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+Follow the steps below to [delete the stacks](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-console-delete-stack.html) you created as a part of this lab:
+
+1. Navigate to the [AWS CloudFormation Console](https://console.aws.amazon.com/cloudformation/).
+2. On the **Stacks** page in the CloudFormation console, select the `resource-return-values` stack.
+3. In the stack details pane, choose **Delete** to delete the stack, and then choose **Delete stack** to confirm.
+4. Repeat steps above to delete the `resource-return-values-challenge` stack you created.
+
+### Conclusion
+
+Great work! You have now learned how to find return values for a specified resource, and how to use them in your templates in other resources using `Ref`,  `Fn::GetAtt` and `Fn::Sub` [intrinsic functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html).
```
Comment (mrinaudo-aws) [inline workshop/content/basics/templates/finding-output-values/_index.md:175]:
```suggestion
Great work! You have now learned how to find return values for a specified resource, and how to use them in your templates in other resources using `Ref`, `Fn::GetAtt`, and `Fn::Sub` [intrinsic functions](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/intrinsic-function-reference.html).
```

File: code/solutions/resource-return-values/resource-return-values-challenge.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: AWS CloudFormation workshop - Resource Dependencies Challenge Lab (uksb-1q9p31idr).
```
Comment (mrinaudo-aws) [inline code/solutions/resource-return-values/resource-return-values-challenge.yaml:3]:
```suggestion
Description: AWS CloudFormation workshop - Resource Return Values Challenge Lab (uksb-1q9p31idr).
```

File: code/solutions/resource-return-values/resource-return-values.yaml [hunk 4]
```diff
@@ -0,0 +1,34 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: AWS CloudFormation workshop - Resource Return Values Lab with DependsOn attribute (uksb-1q9p31idr).
```
Comment (mrinaudo-aws) [inline code/solutions/resource-return-values/resource-return-values.yaml:3]:
```suggestion
Description: AWS CloudFormation workshop - Resource Return Values Lab (uksb-1q9p31idr).
```

File: code/workspace/resource-return-values/resource-return-values-challenge.yaml [hunk 5]
```diff
@@ -0,0 +1,49 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: AWS CloudFormation workshop - Resource Dependencies Challenge Lab (uksb-1q9p31idr).
```
Comment (mrinaudo-aws) [inline code/workspace/resource-return-values/resource-return-values-challenge.yaml:3]:
```suggestion
Description: AWS CloudFormation workshop - Resource Return Values Lab (uksb-1q9p31idr).
```

File: code/workspace/resource-return-values/resource-return-values.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+AWSTemplateFormatVersion: "2010-09-09"
+
+Description: AWS CloudFormation workshop - Resource Return Values Lab with DependsOn attribute (uksb-1q9p31idr).
```
Comment (mrinaudo-aws) [inline code/workspace/resource-return-values/resource-return-values.yaml:3]:
```suggestion
Description: AWS CloudFormation workshop - Resource Return Values Lab (uksb-1q9p31idr).
```

File: workshop/content/basics/templates/finding-output-values/_index.md [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+      Tags:
+        - Key: Purpose
+          Value: AWS CloudFormation Workshop
+
+  S3BucketPolicy:
+    Type: AWS::S3::BucketPolicy
+    Properties:
+      Bucket: !Ref S3Bucket
+      PolicyDocument:
+        Version: 2012-10-17
```
Comment (mrinaudo-aws) [inline workshop/content/basics/templates/finding-output-values/_index.md:60]:
```suggestion
        Version: "2012-10-17"
```

File: workshop/content/basics/templates/finding-output-values/_index.md [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+        Statement:
+          - Action:
+              - 's3:*'
```
Comment (mrinaudo-aws) [inline workshop/content/basics/templates/finding-output-values/_index.md:63]:
```suggestion
              - s3:*
```

File: code/solutions/resource-return-values/resource-return-values-challenge.yaml [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  Ec2Instance:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: t2.micro
+      SecurityGroups:
+        - !Ref InstanceSecurityGroup
+      Tags:
+        - Key: Name
+          Value: Resource-dependencies-workshop
```
Comment (mrinaudo-aws) [inline code/solutions/resource-return-values/resource-return-values-challenge.yaml:20]:
```suggestion
          Value: Resource-return-values-workshop
```

File: code/solutions/resource-return-values/resource-return-values-challenge.yaml [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    Properties:
+      GroupDescription: Allow http to client host
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: 0.0.0.0/0
+      Tags:
+        - Key: Name
+          Value: Resource-dependencies-workshop
```
Comment (mrinaudo-aws) [inline code/solutions/resource-return-values/resource-return-values-challenge.yaml:38]:
```suggestion
          Value: Resource-return-values-workshop
```

File: code/workspace/resource-return-values/resource-return-values-challenge.yaml [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+
+Resources:
+  Ec2Instance:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: t2.micro
+      Tags:
+        - Key: Name
+          Value: Resource-dependencies-workshop
```
Comment (mrinaudo-aws) [inline code/workspace/resource-return-values/resource-return-values-challenge.yaml:18]:
```suggestion
          Value: Resource-return-values-workshop
```

File: code/workspace/resource-return-values/resource-return-values-challenge.yaml [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+    Properties:
+      GroupDescription: Allow http to client host
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 80
+          ToPort: 80
+          CidrIp: 0.0.0.0/0
+      Tags:
+        - Key: Name
+          Value: Resource-dependencies-workshop
```
Comment (mrinaudo-aws) [inline code/workspace/resource-return-values/resource-return-values-challenge.yaml:36]:
```suggestion
          Value: Resource-return-values-workshop
```

File: workshop/content/basics/templates/finding-output-values/_index.md [hunk 13]
```diff
@@ -0,0 +1,175 @@
...
+  Ec2Instance:
+    Type: AWS::EC2::Instance
+    Properties:
+      ImageId: !Ref LatestAmiId
+      InstanceType: t2.micro
+      SecurityGroups:
+        - !Ref InstanceSecurityGroup
+      Tags:
+        - Key: Name
+          Value: Resource-dependencies-workshop
```
Comment (mrinaudo-aws) [inline workshop/content/basics/templates/finding-output-values/_index.md:139]:
```suggestion
          Value: Resource-return-values-workshop
```

**Review by mrinaudo-aws — CHANGES_REQUESTED** [review]
@dreddy-aws thanks!  Please review/approve/update, and request another review!

**Review by mrinaudo-aws — APPROVED** [review]
lgtm

File: workshop/content/basics/templates/finding-return-values/_index.md [hunk 14]
```diff
@@ -0,0 +1,181 @@
+---
```
Comment (mrinaudo-aws) [inline workshop/content/basics/templates/finding-return-values/_index.md:1]: @dreddy-aws can you actually rename the directory to `resource-return-values` for consistency with the code directories?  You might also need to change the image path in the markdown file.  Thanks!

**Review by mrinaudo-aws — DISMISSED** [review]
lgtm

**Review by rezabekf — APPROVED** [review]
Approved.

**Pull request merged**